lib/models/application.coffee

```diff
@@ -237,24 +267,26 @@ getApplicationModel = (deps, opts) ->
 	# @function
 	# @memberof resin.models.application
 	#
-	# @param {String} name - application name
+	# @param {String|Number} nameOrId - application name (string) or id (number)
 	# @returns {Promise}
 	#
 	# @example
 	# resin.models.application.remove('MyApp');
 	#
 	# @example
+	# resin.models.application.remove(123);
+	#
+	# @example
 	# resin.models.application.remove('MyApp', function(error) {
 	# 	if (error) throw error;
 	# });
 	###
-	exports.remove = (name, callback) ->
-		exports.get(name).then ->
+	exports.remove = (nameOrId, callback) ->
+		getId(nameOrId).then (applicationId) ->
 			return pine.delete
 				resource: 'application'
-				options:
-					filter:
-						app_name: name
+				id: applicationId
+		.catch(treat404AsMissingApplication(nameOrId))
```

Comment:
I think this would be nicer as something like:
```coffee
exports.catch404 =
	code: 'ResinRequestError'
	statusCode: 404

exports.treatAsMissingApplication= (nameOrId) ->
	return (err) ->
		replacementErr = new errors.ResinApplicationNotFound(nameOrId))
		replacementErr.stack = err.stack
		throw replacementErr
...
.catch(catch404, treatAsMissingApplication(nameOrId))
```

This does a few things
* it separates the predicate for what you're catching from the conversion, which means they can be composed separately
* it lets us use bluebird sugar for the predicate, including avoiding the need to rethrow
* it lets us create the error object lazily on demand rather than immediately, meaning if it never happens we don't have to pay the cost of creating that error - notably the cost of constructing a stack trace.